src/bin/lttng-relayd/stream.c

```diff
@@ -1070,6 +1070,11 @@ int stream_update_index(struct relay_stream *stream, uint64_t net_seq_num,
 		goto end;
 	}
 
+	if (!stream->trace_chunk) {
+		ret = 0;
```

Comment:
To answer your question, this should not return success.

There is no legitimate case where a stream's trace chunk would be `NULL` while receiving a data packet. It could only result from an internal error. Hence, `stream->trace_chunk != NULL` can be considered a pre-condition of this function.

I would `assert(stream->trace_chunk)` since the only caller of this function calls `stream_write(...)` also has the same pre-condition.

In the case of `stream_write(...)`, the pre-condition is validated by the fact that the stream has an open output file descriptor (otherwise the function would fail). With the trace chunk infrastructure, it is expected that a stream maintains a reference to any trace chunk through which it created one of its opened fd(s).